pipeline/archivebot/control.py

```diff
@@ -182,42 +206,70 @@ def unregister_pipeline(self, pipeline_id):
     # This function is a thread used to asynchronously ship logs to redis for
     # this job, in a daemonic thread
     def ship_logs(self):
-        while True:
-            entry = self.log_queue.get() #infinite blocking
-
-            if entry['type'] is 'log':
-                try:
-                    with conn(self):
-                        self.log_script(keys=entry['keys'], args=entry['args'])
-                except ConnectionError:
-                    pass
-
-            elif entry['type'] is 'bytes_downloaded':
-                try:
-                    with conn(self):
-                        self.redis.hincrby(entry['ident'], 'bytes_downloaded',
-                                entry['bytes'])
-                except ConnectionError:
-                    pass
-
-            elif entry['type'] is 'item_counts':
-                try:
-                    with conn(self):
-                        self.redis.hincrby(entry['ident'], 'items_downloaded',
-                                entry['items_downloaded'])
-                        self.redis.hincrby(entry['ident'], 'items_queued',
-                                entry['items_queued'])
-                except ConnectionError:
-                    pass
-
-            self.log_queue.task_done()
+        shipping_count = 0
+
+        logger.info('Started log shipper thread with ident={}, thread={}'.format(self.ident, threading.get_ident()))
 
+        with conn(self):
+            with self.redis.pipeline(transaction=False) as pipe:
+                while not (self.ending and self.log_queue.empty()):
+
+                    try:
+                        # Ship a log entry
+                        try:
```

Comment:
I've modified it, and also included the wpull2 changes.  This is what ananiel2 and ananiel3 are running.